framework/include/loops/UpdateDisplacedMeshThread.h

```diff
@@ -34,6 +34,12 @@ class UpdateDisplacedMeshThread : public ThreadedNodeLoop<NodeRange, NodeRange::
   virtual void onNode(NodeRange::const_iterator & nd) override;
 
   void join(const UpdateDisplacedMeshThread & /*y*/);
+  virtual void post() override;
+
+  /**
+   * Whether this just modified the mesh with operator()
+   */
+  bool hasDisplacement() const { return _has_displacement; }
```

Comment:
This is a brittle method and only has meaning right after a call to the `operator()` method. This should either be made private and you friend `DisplacedProblem` (or better yet just the `DisplacedMesh::updateMesh` methods) or provide a key that only `DisplacedProblem` can construct